src/solvers/dg/dg_2d_parallel.jl

```diff
@@ -1,2 +1,309 @@
-# TODO: Taal MPI
+
+# everything related to a DG semidiscretization in 2D using MPI,
+# currently limited to Lobatto-Legendre nodes
+
+
+# TODO: MPI dimension agnostic
+# TODO: MPI, adapt to different real types?
+mutable struct MPICache
+  mpi_neighbor_ranks::Vector{Int}
+  mpi_neighbor_interfaces::Vector{Vector{Int}}
+  mpi_send_buffers::Vector{Vector{Float64}}
+  mpi_recv_buffers::Vector{Vector{Float64}}
+  mpi_send_requests::Vector{MPI.Request}
+  mpi_recv_requests::Vector{MPI.Request}
+  n_elements_by_rank::OffsetArray{Int, 1, Array{Int, 1}}
+  n_elements_global::Int
+  first_element_global_id::Int
+end
+
+
+# TODO: MPI dimension agnostic
+function start_mpi_receive!(mpi_cache::MPICache)
+
+  for (index, d) in enumerate(mpi_cache.mpi_neighbor_ranks)
+    mpi_cache.mpi_recv_requests[index] = MPI.Irecv!(
+      mpi_cache.mpi_recv_buffers[index], d, d, mpi_comm())
+  end
+
+  return nothing
+end
+
+
+# TODO: MPI dimension agnostic
+function start_mpi_send!(mpi_cache::MPICache, mesh, equations, dg, cache)
```

Comment:
I am assuming you did not change anything in the parallel algorithms except refactoring some names etc. If you did, please let me know such that I can re-verify the relevant functions.

Reply:
I didn't change any parallel algorithm on purpose. I just refactored them, using different signatures etc. if necessary.